### PR TITLE
Email stats open rate UI

### PR DESCRIPTION
**Pull request by TimBroddin**
#### Proposed Changes

This PR adds the open rate to the email stats. 

- It introduces the concept of period bound stats & all-time stats. Because of this I refactored the Redux actions and selectors a bit.
- It uses a component similar to `<HighlightCard />` to render the info on top.
- It also fetched the post, so we're able to show the title. While this is being fetched, a spinner is shown.

CleanShot 2023-01-10 at 14 51 58@2x

#### Testing Instructions

* Visit `/stats/email/open/<site>/day/<postid>?flags=newsletter/stats`. You can add the startDate parameter, to load the stats for a specific date.
* Open rate should be at the top

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by JuanLucha** [comment]
Mm... I have been testing and seems like there is something odd. Please, check the video. There I refresh the page (that's why you see the `W` logo in white screen for a moment) and then everything loads while the endpoints retrieve the data but then... the new cards on top keeps the rolling-rolling-rolling-rolling 🛼 😄 

https://user-images.githubusercontent.com/3832570/211612303-63a5e9be-4802-4e80-9d54-f0a955095424.mov

Ping me if you need further help testing this 👍

**Comment by TimBroddin** [comment]
Hey @JuanLucha 

I fixed the loading problem. One thing you'll notice, is that for emails with no stats, total opens will show as 0 and the other ones will be `-`.

CleanShot 2023-01-11 at 12 04 54@2x

This is because the API returns this shape:

CleanShot 2023-01-11 at 12 05 56@2x

I could catch this in this PR, but I think it might be better if @phcp changes this in the API (so that `total_opens` also returns `null` when `total_sends` is `null`).

**Comment by JuanLucha** [comment]
Mm... there is something wrong with the parsing of the title. My post title is `Testing 09-01-23 – Without Autoproxxy` but the screen shows this strange encoding:
Screenshot 2023-01-11 at 13 31 34

**Comment by TimBroddin** [comment]
Fixed it!

File: client/my-sites/stats/stats-email-open-top-row/index.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Gridicon } from '@automattic/components';
+import { Icon } from '@wordpress/icons';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import {
+	getAlltimeStats,
+	isRequestingAlltimeEmailStats,
```
Comment (JuanLucha) [inline client/my-sites/stats/stats-email-open-top-row/index.jsx:8]: Something weird happen to the indentation here...

File: client/my-sites/stats/stats-email-open-top-row/top-card.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Card, ShortenedNumber, Spinner } from '@automattic/components';
+
+/* This is a very stripped down version of HighlightCard
+ * HighlightCard doens't support non-numeric values
+ * */
+
+const TopCardValue = ( { value, isLoading } ) => {
+	const isNumber = Number.isFinite( value );
+
```
Comment (JuanLucha) [inline client/my-sites/stats/stats-email-open-top-row/top-card.jsx:9]: The indentation in this file seems odd too.

**Review by JuanLucha — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7727007

Thank you @TimBroddin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7727007).

**Pull request merged**